src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -115,6 +115,13 @@ private Expression ParseExpressionSegment()
 			// MSDN C# "Operator precedence and associativity"
 			// https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/
 
+			if (_arguments.Settings.LambdaExpressions)
+			{
+				var lambdaExpr = ParseLambdaExpression();
+				if (lambdaExpr != null)
+					return lambdaExpr;
+			}
```

Comment:
It is maybe safer to put this section inside the `try/catch` below to catch any `InvalidOperationException`?

Reply:
The try/catch was handled in `ParseLambdaExpression`, but if I change that try/catch to only catch `ParseException`, I'll put this section in the other try/catch.